scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -301,252 +321,266 @@
+                                          :pod/running (kill-pod api-client name cook-expected-state-dict pod)
+                                          ; We shouldn't hit these unless we get a database rollback.
+                                          :pod/succeeded (kill-pod-in-weird-state compute-cluster pod-name
+                                                                                  nil k8s-actual-state-dict)
+                                          ; Unlike the other :pod/unknown states, no datomic state to update.
+                                          :pod/unknown (kill-pod-in-weird-state compute-cluster pod-name
                                                                                 nil k8s-actual-state-dict)
-                                        ; Unlike the other :pod/unknown states, no datomic state to update.
-                                        :pod/unknown (kill-pod-in-weird-state compute-cluster pod-name
-                                                                              nil k8s-actual-state-dict)
-                                        ; This can occur in testing when you're e.g., blowing away the database.
-                                        ; It will go through :missing,:missing and then be deleted from the map.
-                                        ; TODO: May be evidence of a bug where we process pod changes when we're starting up.
-                                        ; Currently occurs because kill's can race ahead of launches, we kill something that has
-                                        ; been added to datomic, but hasn't been submitted to k8s yet.
-                                        :pod/waiting (kill-pod api-client cook-expected-state-dict pod)))]
-      (when-not (cook-expected-state-equivalent? cook-expected-state-dict new-cook-expected-state-dict)
+                                          ; This can occur in testing when you're e.g., blowing away the database.
+                                          ; It will go through :missing,:missing and then be deleted from the map.
+                                          ; TODO: May be evidence of a bug where we process pod changes when we're starting up.
+                                          ; Currently occurs because kill's can race ahead of launches, we kill something that has
+                                          ; been added to datomic, but hasn't been submitted to k8s yet.
+                                          :pod/waiting (kill-pod api-client name cook-expected-state-dict pod)))]
+        ; We do not want to gate whether we update on cook-expected-state-equivalent.
+        ; cook-expected-state-equivalent? is intended to identify when the states are different in a way that
+        ; may need reprocessing in the state machine.
+        ; However, the state may have changed in a way we want to capture. (e.g., metric changes) and we want to
+        ; capture that change.
```

Comment:
Can we remove one of the two occurrences of "we want to capture" from this sentence? Seems redundant.

Reply:
Reworded.